src/riscv/lib/src/interpreter/float.rs

```diff
@@ -609,6 +614,77 @@ const fn status_to_bits(status: Status) -> u8 {
     status.bits().reverse_bits() >> 3
 }
 
+enum SignInjection {
+    Identity,
+    Negate,
+    Xor,
+}
+
+fn f64_sign_inject(
+    icb: &mut impl ICB,
+    rs1: FRegister,
+    rs2: FRegister,
+    rd: FRegister,
+    manip: SignInjection,
+) {
+    let rval1_fvalue = icb.fregister_read(rs1);
+    let rval2_fvalue = icb.fregister_read(rs2);
+
+    let rval1_xvalue = icb.xvalue_from_fvalue(rval1_fvalue);
+    let rval2_xvalue = icb.xvalue_from_fvalue(rval2_fvalue);
+
+    let bit_mask_rs1 = icb.xvalue_of_imm(F64_SIGNLESS_VALUE_MASK);
+    let bitmask_rs2 = icb.xvalue_of_imm(F64_SIGN_BIT_MASK);
+
+    let rval2_xvalue_sign = rval2_xvalue.and(bitmask_rs2, icb);
+
+    let rval2_xvalue_sign = match manip {
+        SignInjection::Identity => rval2_xvalue_sign,
```

Comment:
In this case you can just use [fcopysign](https://docs.rs/cranelift-codegen/latest/cranelift_codegen/ir/trait.InstBuilder.html#method.fcopysign).